dashboard/app/models/pd/workshop_organizer_report.rb

```diff
@@ -0,0 +1,120 @@
+class Pd::WorkshopOrganizerReport
+
+  # Construct a report row for each workshop for each organizer
+  def self.generate_organizer_report(organizer)
+    closed_workshops = Pd::Workshop.in_state(Pd::Workshop::STATE_ENDED)
+
+    if organizer.admin?
+      return closed_workshops.all.map do |workshop|
+        generate_organizer_report_row(workshop)
+      end
+    end
+
+    closed_workshops.organized_by(organizer).all.map do |workshop|
+      generate_organizer_report_row(workshop)
+    end
+  end
+
+  def self.generate_organizer_report_row(workshop)
+    teachers = Pd::Attendance.for_workshop(workshop).distinct_teachers
+    plp = ProfessionalLearningPartner.find_by_contact_id(workshop.organizer.id)
+    section_url = workshop.section ?
+      CDO.code_org_url("/teacher-dashboard#/sections/#{workshop.section.id}", 'http:') : nil
+    payment_type =
+      if plp
+        plp.urban? ? 'PLP Urban' : 'PLP Non-urban'
+      elsif workshop.course == Pd::Workshop::COURSE_CSF
+        'CSF Facilitator'
+      else
+        nil
+      end
+
+    qualified, payment_amount = calculate_payment(workshop, teachers, plp)
+    {
+      organizer_name: workshop.organizer.name,
+      organizer_id: workshop.organizer.id,
+      organizer_email: workshop.organizer.email,
+      workshop_dates: workshop.sessions.map(&:formatted_date).join(' '),
+      workshop_type: workshop.workshop_type,
+      section_url: section_url,
+      facilitators: workshop.facilitators.map(&:name).join(', '),
+      workshop_name: workshop.friendly_name,
+      course: workshop.course,
+      subject: workshop.subject,
+      num_teachers: teachers.count,
+      days: workshop.sessions.count,
+      payment_type: payment_type,
+      qualified: qualified,
+      payment_amount: payment_amount
+    }
+  end
+
+  def self.calculate_payment(workshop, teachers, plp)
```

Comment:
Payment calculation belongs in its own class, and arguably multiple classes.  At the risk of nitpicking, it seems to me that reports should display results more than calculate those results in the first place.

Happy to talk more about it offline